internal/pkg/cli/deploy.go

```diff
@@ -547,11 +548,48 @@ func (o *deployOpts) askNames() error {
 		return nil
 	}
 
-	names, err := o.sel.Workloads("Select a service or job in your workspace", "")
+	names, err := o.sel.Workloads("Select one or more services or jobs in your workspace.", "")
 	if err != nil {
 		return fmt.Errorf("select service or job: %w", err)
 	}
-	o.workloadNames = names
+	if len(names) == 1 {
+		o.workloadNames = names
+		return nil
+	}
+
+	// Select workload priority by repeatedly prompting with a multiselect until the list of options is depleted.
+	options := names
+	taggedOptions := make([]string, 0, len(options))
+	currentPriority := 1
+	for len(options) > 0 {
+		deploymentGroupIndex := "next"
+		if currentPriority == 1 {
+			deploymentGroupIndex = "first"
+		}
+		selectedNames, err := o.prompt.MultiSelect(
+			fmt.Sprintf("Which workload(s) would you like to deploy in your %s deployment group?", deploymentGroupIndex),
+			"These workloads will be deployed together.",
+			append(options, optionAllRemainingWorkloads),
+			prompt.RequireMinItems(1),
+			prompt.WithFinalMessage(fmt.Sprintf("Group %d", currentPriority)),
+		)
+		if err != nil {
+			return fmt.Errorf("select workloads for deployment group: %w", err)
+		}
+		if slices.Contains(selectedNames, optionAllRemainingWorkloads) {
+			selectedNames = options
+		}
+
+		// Tag all selected workloads with the given priority.
+		for _, name := range selectedNames {
+			taggedOptions = append(taggedOptions, fmt.Sprintf("%s/%d", name, currentPriority))
+		}
```

Comment:
Maybe we should make `workloadNames` a nested slice `[][]string` in order to make this more robust? I'm not a huge fan of tagging strings because it means user input has some effect on the code, even though this shouldn't create any security issue.

Reply:
How do you mean? This has implications for flags--is there a good way to structure a nested slice with cobra flags? Otherwise there's not a great way to specify order that maintains backwards compatibility.

Reply:
I discussed with Wanxian about this and I get now why you have this concern. The other alternative I can think of is that we add an extra variable to `deployOpts` which contains cached information about the priority of deploying different workloads, which this would be `[][]string` or `map[int][]string`. Then we'd use this cached variable for deployment so that `workloadNames` can be the variable set by user input.

Wanxian did point out a drawback of this approach though, being that we then need to make sure that the `workloadNames` and cached information stay equivalent, but in this case I think it should be manageable. Let me know what you think about this approach.